File: junit5/src/test/java/io/micronaut/test/junit5/RequiresTest.java
package io.micronaut.test.junit5;

import io.micronaut.context.annotation.Requires;
import io.micronaut.test.junit5.annotation.MicronautTest;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

@MicronautTest
@Requires(property = "does.not.exist")
class RequiresTest {

    @Test
    void testNotExecuted() {
        Assertions.fail("Should never be executed");
    }
}

File: test-kotest5/src/test/kotlin/io/micronaut/test/kotest5/RequiresTest.kt

package io.micronaut.test.kotest5

import io.kotest.assertions.fail
import io.kotest.core.spec.style.BehaviorSpec
import io.micronaut.context.annotation.Requires
import io.micronaut.test.extensions.kotest5.annotation.MicronautTest

@MicronautTest
@Requires(property = "does.not.exist")
class RequiresTest: BehaviorSpec({

    given("a test with requires") {
        fail("Should never be executed")
    }

})
